### DISCUSSION TITLE
How to apply specific scalings to a specific window?

### DISCUSSION THREAD

**Discussion by Wanye-7300** [hyprwm/Hyprland#8712] [Q&A]
> Translated from Chinese

My monitor's scaling is 1.6 and the resolution is 2560x1440. But when I opened Minecraft, the game showed a resolution of 3200x1800. This means that my game has been scaled as well. Since this usually affects frames per second, I'd like to set scaling of 1 for this window separately to revert back to a resolution of 2560x1440. Is there any way I can do this or the same? Appreciate!

**Comment by vaxerski** [comment]
considering that, I assume it's running in native wayland. No, there is no way to get around this, unless the app itself supports the fractional scaling protocol.

**Reply by Wanye-7300** [reply]
> considering that, I assume it's running in native wayland. No, there is no way to get around this, unless the app itself supports the fractional scaling protocol.

> GLFW uses the [fractional-scale-v1](https://wayland.app/protocols/fractional-scale-v1) protocol to implement fine-grained framebuffer scaling. If the running compositor does not support this protocol, the [GLFW_SCALE_FRAMEBUFFER](https://www.glfw.org/docs/latest/group__window.html#gaa5a9c6b4722670fd33d6e8a88f2e21bc) window hint will only be able to scale the framebuffer by integer scales.

Minecraft uses the library of GLFW, so I think that it should support fractional scaling.

**Reply by vaxerski** [reply]
depends. Commercial apps have a history of using very outdated dependencies. At any rate, this is not a hyprland issue. All other apps scale fractionally :)
